src/index.js

```diff
@@ -140,7 +141,10 @@ class ServerlessPlugin {
    */
   parseEnvFiles(envFileNames) {
     const envVarsArray = envFileNames.map((fileName) => {
-      const parsed = dotenv.config({ path: fileName });
+      const parsed = dotenv.config({
+        path: fileName,
+        override: this.config.envFileOnly,
+      });
```

Comment:
[`dotenv.config()` accepts other arguments as well](https://github.com/motdotla/dotenv#config). I would prefer to just add support for passing configs down. E.g.

```
custom:
  dotenv:
    dotenv:
      config:
        encoding: 'utf-16'
        override: true
```

```suggestion
      const parsed = dotenv.config({
        ...this.config.dotenv.config,
        path: fileName,
      });
```